File: UpravljanjeSZ/test/javascript/pages/zgrade/ZgradaPage.js
//konstruktor stranice
var ZgradaPage = function () { }
var utils = require('../utils.js');
// Opis polja i metoda
ZgradaPage.prototype = Object.create({}, {

    // h2 poruka Zgrada za asertaciju da sam na stranici zgrada
    textMsg: {
        get: function () {
            return utils.waitForElementPresence(by.xpath("//div[@class='container']/h2"),10000).getText();
        }
    },
    // adresa
    adresa: {
        get: function () {
            return utils.waitForElementPresence(by.xpath("/html/body/app-root/app-zgrada/div/div[1]/h5[1]"),10000).getText();
        }
    },
    // predsednik
    predsednik: {
        get: function () {
            return utils.waitForElementPresence(by.xpath("//html/body/app-root/app-zgrada/div/div[1]/h5[2]"),10000).getText();
        }
    },
    // predsedniklink 
    predsednikLink: {
        get: function () {
            return utils.waitForElementPresence(by.partialLinkText("Gospodin Predsednik"),10000);
        }
    },
    //SELEKT
    prikaz: {
        get: function() {
            return utils.waitForElementPresence(by.id('prikaz'), 10000).getText();
        },
        set: function(value) {
            this.prikaz.element(by.cssContainingText('option', value)).click();
        }
    },
    // meni sa tabovima
    menuTab: {
        get: function () {
            return utils.waitForElementPresence(by.xpath("//ul[@class='nav nav-tabs']"),10000);
        }
    },
    // tab stanovi
    stanoviTab: {
        get: function () {
            return utils.waitForElementPresence(by.linkText("Stanovi"),10000);
        }
    },
    // tab obavestenja
    obavestenjaTab: {
        get: function () {
            return utils.waitForElementPresence(by.linkText("Obavestenja"),10000);
        }
    },
    // tab predlozi tacke dnevnog reda
    predloziTab: {
        get: function () {
            return utils.waitForElementPresence(by.linkText("Predlozi tacke dnevnog reda"),10000);
        }
    },
    // tab sastanci skupstine
    sastanciTab: {
        get: function () {
            return utils.waitForElementPresence(by.linkText("Sastanci skupstine"),10000);
        }
    },
    // tab kvarovi
    kvaroviTab: {
        get: function () {
            return utils.waitForElementPresence(by.linkText("Kvarovi"),10000);
        }
    },
    // TABELA
    // cela tabela sa stanovima
    stanoviTable: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//table[@class='table table-hover']"), 10000);
        }
    },
    // red mog stana izvucen iz tabele po vlasniku
    getStanIzTabele: {
        value: function(vlasnik) {
            return utils.waitForElementPresence(by.xpath('//*[contains(text(),"' + vlasnik + '")]/../..'),10000);
        }
    },
    // vlasnik link u tabeli
    linkVlasnik: {
        get: function() {
            return utils.waitForElementPresence(by.xpath('/html/body/app-root/app-zgrada/div/div[2]/app-stanovi/div/div[1]/table/tbody/tr[1]/td[1]/a'),10000);
        },
    },
    // dobavljanje vlasnik i stanari dugmeta
    btnVlasnikStanar: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("/html/body/app-root/app-zgrada/div/div[2]/app-stanovi/div/div[1]/table/tbody/tr[1]/td[2]/a"),10000);
        },
    },
    // dobavljanje cekboksa na stranici kvarovi
    cbZavrseniKvarovi: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//input[@type='checkbox']"),10000);
        },
    }

});

// Export klase
module.exports = ZgradaPage;

File: UpravljanjeSZ/main/java/pages/zgrade/ZgradaPage.java
package pages.zgrade;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

import pages.helpers.Utility;

public class ZgradaPage {
	private WebDriver driver;

	// konstruktor stranice
	public ZgradaPage(WebDriver driver) {
		this.driver = driver;
	}

	// h2 poruka Zgrada za asertaciju da sam na stranici zgrada
	public String getTextMsg() {
		return Utility.waitForElementPresence(driver, By.xpath("//div[@class='container']/h2"), 10).getText();
	}

	// adresa
	public String getTextAdresa() {
		return Utility.waitForElementPresence(driver, By.xpath("/html/body/app-root/app-zgrada/div/div[1]/h5[1]"), 10)
				.getText();
	}

	// predsednik
	public String getTextPredsednik() {
		return Utility.waitForElementPresence(driver, By.xpath("/html/body/app-root/app-zgrada/div/div[1]/h5[2]"), 10)
				.getText();
	}

	// predsedniklink
	public WebElement getLinkPredsednik() {
		return Utility.waitForElementPresence(driver, By.partialLinkText("Gospodin Predsednik"), 10);
	}

	// SELEKT
	public Select getPrikaz() {
		return new Select(Utility.waitForElementPresence(driver, By.id("prikaz"), 10));
	}

	public void setPrikaz(String value) {
		this.getPrikaz().selectByVisibleText(value);
	}
	// meni sa tabovima
	public WebElement getTabMenu() {
		return Utility.waitForElementPresence(driver, By.xpath("//ul[@class='nav nav-tabs']"), 10);
	}

	// tab stanovi
	public WebElement getStanoviTab() {
		return Utility.waitForElementPresence(driver, By.linkText("Stanovi"), 10);
	}

	// tab obavestenja
	public WebElement getObavestenjaTab() {
		return Utility.waitForElementPresence(driver, By.linkText("Obavestenja"), 10);
	}

	// tab predlozi tacke dnevnog reda
	public WebElement getPredloziTab() {
		return Utility.waitForElementPresence(driver, By.linkText("Predlozi tacke dnevnog reda"), 10);
	}

	// tab sastanci skupstine
	public WebElement getSastanciTab() {
		return Utility.waitForElementPresence(driver, By.linkText("Sastanci skupstine"), 10);
	}

	// tab kvarovi
	public WebElement getKvaroviTab() {
		return Utility.waitForElementPresence(driver, By.linkText("Kvarovi"), 10);
	}

	// TABELA
	// cela tabela sa stanovima
	public WebElement getStanoviTable() {
		return Utility.waitForElementPresence(driver, By.xpath("//table[@class='table table-hover']"), 10);
	}

	// red mog stana izvucen iz tabele po vlasniku
	public WebElement getStanIzTabele(String ime) {
		return Utility.waitForElementPresence(driver, By.xpath("//*[contains(text(),\"" + ime + "\")]/../.."), 10);
	}

	// metoda za proveru da li stan postoji u tabeli, po imenu vlasnika
	public boolean isStanPresent(String ime) {
		return Utility.isPresent(driver, By.xpath("//*[contains(text(),\"" + ime + "\")]/../.."));
	}

	// vlasnik link u tabeli
	public WebElement getLinkVlasnik() {
		return Utility.waitForElementPresence(driver,
				By.xpath("/html/body/app-root/app-zgrada/div/div[2]/app-stanovi/div/div[1]/table/tbody/tr[1]/td[1]/a"),
				10);
	}

	// dobavljanje vlasnik i stanari dugmeta
	public WebElement getBtnVlasnikStanar() {
		return Utility.waitForElementPresence(driver,
				By.xpath("/html/body/app-root/app-zgrada/div/div[2]/app-stanovi/div/div[1]/table/tbody/tr[1]/td[2]/a"),
				10);
	}
	 // dobavljanje cekboksa na stranici kvarovi
	public WebElement getCbZavrseniKvarovi() {
		return Utility.waitForElementPresence(driver,
				By.xpath("//input[@type='checkbox']"),
				10);
	}
   
}
